src/capability.rs

```diff
@@ -61,25 +69,35 @@ impl NativeCapability {
         Ok(NativeCapability {
             capid,
             plugin,
+            binding_name: binding_name.unwrap_or("default".to_string()),
             library: Some(library),
         })
     }
 
-    /// If you know ahead of time that you want a particular capability provider to be a compile-time
-    /// dependency, you can create your own provider instance and pass it to this function
-    pub fn from_instance(instance: impl CapabilityProvider) -> Result<Self> {
+    /// This function is to be used for _capability embedding_. If you are building a custom
+    /// waSCC host and have a fixed set of capabilities that you want to always be available
+    /// to actors, then you can declare a dependency on the capability provider, enable
+    /// the `static_plugin` feature, and provide an instance of that provider. Be sure to check
+    /// that the provider supports capability embedding.    
+    pub fn from_instance(
+        instance: impl CapabilityProvider,
+        binding_name: Option<String>,
+    ) -> Result<Self> {
         let capid = instance.capability_id();
         Ok(NativeCapability {
             capid: capid.to_string(),
             plugin: Box::new(instance),
+            binding_name: binding_name.unwrap_or("default".to_string()),
             library: None,
         })
     }
 
+    /// Returns the capability ID (namespace) of the provider
     pub fn id(&self) -> String {
         self.capid.clone()
     }
 
+    /// Returns the human-friendly name of the provider
     pub fn name(&self) -> String {
         self.plugin.name().to_string()
```

Comment:
I see that you `clone()` the capability ID just above this, does `to_string()` in Rust effectively clone the String, or should this be cloned as well? I'd assume if it was previously an &str then it's not the same reference, just curious.

Reply:
Idiomatically it's generally safer to call `to_string()` if the intent is to convert something into an owned string, since `to_string` will clone if the thing is already a string, and it'll produce a new string if the call site is an `&str`. I'll refactor this for consistency